AppSettingsManagerAPI/AppSettingsManagerApi.Domain/MySql/IPermissionRepository.cs

```diff
@@ -0,0 +1,17 @@
+using AppSettingsManagerApi.Model;
+using AppSettingsManagerApi.Model.Requests;
+using Permission = AppSettingsManagerApi.Model.Permission;
+
+namespace AppSettingsManagerApi.Domain.MySql;
+
+public interface IPermissionRepository
+{
+    Task<Permission> GetPermission(string userId, string settingGroupId);
+    Task<Permission> CreatePermission(CreatePermissionRequest request);
+    Task<Permission> UpdatePermission(UpdatePermissionRequest request);
+    Task<Permission> PermissionRequestResponse(PermissionRequestResponse response);
+    Task<Permission> DeletePermission(string userId, string settingGroupId);
+
+    Task CheckPermission(string userId, string password, string settingGroupId,
```

Comment:
It might be better in the future to have this actually return true/false instead of just throwing an exception but this is fine for now